### ISSUE TITLE
RTER Pending Expense - RTER Pending expense is not moved to new report when it's approved

### ISSUE THREAD

**Issue by isagoico** [Expensify/App#47511]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v9.0.20-6

**Reproducible in staging?:** Yes 
**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://github.com/Expensify/Expensify/issues/403524#issuecomment-2290080995

**Email or phone of affected tester (no customers):** applausetester+pendingreceipt@applause.expensifail.com

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Issue reported by:** Applause - Internal team
**Slack conversation:**

## Action Performed:
*Preconditions: a workspace with scheduled submit enabled, a employee and admin and the employee has a company card assigned.* 

1. In an account that has a domain controlled company card (not Expensify Card)
2. Submit a expense via scan in the workspace chat
3. Wait for the smartscan to finish
4. Verify the expense has pending status
5. In the workspace chat, submit a manual expense
6. Submit the report
7. As the approver - Approve the report

## Expected Result:
The pending expense should be moved to a new Report. 

## Actual Result:
The pending expense is not moved to a new report and it still has the Pending Status.

Below are the details of a report with this issue: 
Account: applausetester+pendingreceipt@applause.expensifail.com 
Report ID: 8189390567239041
Transaction ID: 6273679857286185

## Workaround:
N/A 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![image](https://github.com/user-attachments/assets/3d09d9d2-669f-4b56-a691-b90fdfe4439b)

![image](https://github.com/user-attachments/assets/a52139e5-812f-4730-8925-de42dea3ef7d)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
Going to drop off this one, Gents. I'm sure you know what you're doing :)

**Comment by yuwenmemon** [comment]
Looking today

**Comment by yuwenmemon** [comment]
After some [further discussion in Slack](https://expensify.slack.com/archives/CC7NECV4L/p1724250558796739?thread_ts=1724057494.873569&cid=CC7NECV4L) we were conflating this with the other Pending CC Transaction state, which is affecting the customer.

This one might actually benefit from waiting a bit while we get the RTER Violation in Auth (cc @deetergp), since we would be able to keep all logic there and have it be 1:1:1.

**Comment by yuwenmemon** [comment]
Waiting on https://github.com/Expensify/Auth/pull/11972 before implementing this in Auth

**Comment by yuwenmemon** [comment]
Still waiting on https://github.com/Expensify/Auth/pull/11972

**Comment by yuwenmemon** [comment]
Still on HOLD

**Comment by yuwenmemon** [comment]
Still on HOLD

**Comment by deetergp** [comment]
Just a heads up that https://github.com/Expensify/Auth/pull/11972 was deployed to production a few days ago.

**Comment by yuwenmemon** [comment]
Niiice! Thanks for the heads up!

**Comment by yuwenmemon** [comment]
Relevant DD section: https://docs.google.com/document/d/1zJqlTe_RajuBtfQYvbMx8PpXgA9CEnUGVyuqZihQ-ok/edit#heading=h.xp9szvgb3nej

**Comment by yuwenmemon** [comment]
Based on [this conversation](https://expensify.slack.com/archives/C03TQ48KC/p1726095204771559) the full ramp-up for violations will take longer than just the implementation of the RTER violation we had been waiting on, so I'm just going to go with the original way we had it in the design doc.

**Comment by yuwenmemon** [comment]
@war-in I'm looking to do something similar to what you and @robertjchen did for Held Requests with [this PR](https://github.com/Expensify/App/pull/42573) (and [Issue](https://github.com/Expensify/App/issues/41652)), except with Expenses that have the "Receipt pending match with card transaction" _Violation_ - are you interested in giving this a look?

I can work on the back-end components for ya.

**Comment by yuwenmemon** [comment]
Thanks @war-in!

Okay so based on the _continuing_ conversation [here](https://expensify.slack.com/archives/C03TQ48KC/p1726138686359389?thread_ts=1726095204.771559&cid=C03TQ48KC), I think ideally we still wait to make the back-end changes until we have Violations fully in Auth. - which seems like it will be a couple of weeks now at the minimum...

That being said, what we'd want to do in the client is treat all transactions with a [Pending RTER Violation](https://github.com/Expensify/App/blob/128cd29aa1d979e61e1ca56c9273e02cc7f5933c/src/libs/TransactionUtils/index.ts#L566-L571) like they're held. Does that make sense @zfurtak?

**Comment by zfurtak** [comment]
Hi 👋
Sorry but I don't have access to the linked chat. 
> That being said, what we'd want to do in the client is treat all transactions with a [Pending RTER Violation](https://github.com/Expensify/App/blob/128cd29aa1d979e61e1ca56c9273e02cc7f5933c/src/libs/TransactionUtils/index.ts#L566-L571) like they're held. Does that make sense @zfurtak?

So similarly to the Held Requests, after approving we want to transfer them to to a new report?

**Comment by zfurtak** [comment]
hi @isagoico, could you tell me how can I add this card to reproduce the issue? 😊

**Comment by yuwenmemon** [comment]
> Sorry but I don't have access to the linked chat.

Yeah, sorry about that. It's an internal one. In that thread, I was asking about our back-end logic for violations. We're currently in the middle of moving them from one system to another. Based on some recent developments in that discussion, we wanted to hold the back-end changes until that's done.

Now, this puts us at a bit of a crossroads. We can hold off until the back-end is ready to go (maybe about a month, hopefully less) or we can move forward on these front-end portions without being able to reliably test for the moment. Up to us.

**Comment by yuwenmemon** [comment]
As for reproducing, @isagoico I actually don't know how you set that up 😅 - but if you wanted to import a card into your Expensify account you can follow the instructions here: https://help.expensify.com/articles/expensify-classic/connect-credit-cards/Personal-Credit-Cards

**Comment by zfurtak** [comment]
@yuwenmemon 
If this issue is not urgent I would wait for the backend to be ready, as the process of writing frontend code without the backend part is usually slower and may produce some issues due to lack of testing. What do you think? 🤔

**Comment by yuwenmemon** [comment]
Yep, that sounds good to me I will place this on HOLD once more.

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Still waiting on violations in auth

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Holding on other projects.

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by garrettmknight** [comment]
[Just adding the issue that this is held on for reference](https://github.com/Expensify/Expensify/issues/393838)

**Comment by yuwenmemon** [comment]
Un-holding as violations in Auth are done now.

**Comment by JmillsExpensify** [comment]
Nice, thank you!

**Comment by yuwenmemon** [comment]
I took a look at this today, and some questions came up. I'm asking @pecanoro for her advice on how to proceed best with getting violations in Auth for use within Auth.

**Comment by yuwenmemon** [comment]
Okay so we're still holding on this just a touch. Rocio let me know that we can't use `Violations::get` yet on production code as it's still not accurate with its PHP version. We fixing the diff here: https://github.com/Expensify/Expensify/issues/457686

**Comment by yuwenmemon** [comment]
Looks like the issue above is in Review so we're close!

**Comment by yuwenmemon** [comment]
Talked with @pecanoro IRL during onshore and it looks like there are some still issues with the RTER violation diff - so still in a holding pattern on this one for now.

**Comment by JmillsExpensify** [comment]
Added the hold

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
On hold

**Comment by JmillsExpensify** [comment]
Still on hold for violations

**Comment by JmillsExpensify** [comment]
Same 'ole story

**Comment by JmillsExpensify** [comment]
Same same

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by JmillsExpensify** [comment]
Still holding

**Comment by JmillsExpensify** [comment]
Same same

**Comment by JmillsExpensify** [comment]
@deetergp do you have bandwidth to pick this one up since Yuwen is out?

**Comment by deetergp** [comment]
Yeah, I should be able to take it over 👍

**Comment by deetergp** [comment]
No update

**Comment by JmillsExpensify** [comment]
Same

**Comment by deetergp** [comment]
I will get to this soon, just trying to put FIXVIOLATIONS system messages to 🛏️

**Comment by deetergp** [comment]
No update

**Comment by deetergp** [comment]
No update

**Comment by JmillsExpensify** [comment]
Same

**Comment by deetergp** [comment]
Will try to get to this tomorrow afternoon.

**Comment by deetergp** [comment]
> We should try to get this one wrapped up. It's holding up a regression test from going live.

Agreed! I'll aim to at least get a PR started for this tomorrow.

**Comment by deetergp** [comment]
That is clutch—thanks for the tip!

**Comment by deetergp** [comment]
Same as last week, between fixing the batch and holidays.

**Comment by deetergp** [comment]
Spent more time on this today and I think everything is working as intended. I'm going to look at it with fresh eyes in the AM, time the query, then send it out for review.

**Comment by deetergp** [comment]
In final review. Should go out tomorrow!

**Comment by JmillsExpensify** [comment]
Woo, thank you!

**Comment by deetergp** [comment]
This didn't auto close because I had mislabeled the PR details. This hit prod earlier today, so I'm closing it out!

**Comment by deetergp** [comment]
@isagoico After this went out, it was pointed out to me by @sakluger that when submitting manually, the pending expense SHOULD get submitted along with the rest. The correct behavior is that it should be held back only when it is automatically submitted (aka harvested). I've got a PR that addresses that [here](https://github.com/Expensify/Auth/pull/20094) that has been approved and merged but not yet deployed.

**Comment by deetergp** [comment]
This should have been auto closed. The code has been in production for a while.
